src/EditorFeatures/Core/Tagging/AbstractAsynchronousTaggerProvider.TagSource.cs

```diff
@@ -73,12 +73,22 @@ private sealed partial class TagSource : ForegroundThreadAffinitizedObject
             /// </summary>
             private readonly AsyncBatchingWorkQueue<NormalizedSnapshotSpanCollection> _normalPriTagsChangedQueue;
 
-            private readonly ReferenceCountedDisposable<TagSourceState> _tagSourceState = new(new TagSourceState());
```

Comment:
this basically just needed to wrap a CTS + queue in a threadsafe fashion.  However, we can just use a normal AsyncBatchingWorkQueue (already threadsafe) and pull the CTS up to here.  we can be threadsafe on the CTS as all the places that use it already have a place they run on the UI thread, so they can be appropriately safe wrt to DIsposal (also on the UI thread).